pkg/archive/archive.go

```diff
@@ -193,10 +193,10 @@ func ExtractionMethod(ext string) func(context.Context, string, string) error {
 	switch ext {
 	case ".jar", ".zip", ".whl":
 		return ExtractZip
-	case ".gz":
-		return ExtractGzip
 	case ".apk", ".gem", ".tar", ".tar.bz2", ".tar.gz", ".tgz", ".tar.xz", ".tbz", ".xz":
```

Comment:
Would it make sense to leave a comment here to indicate that added compression cases should go below this one?

Reply:
Probably. The ordering can definitely cause issues if one extension is a substring of another.

Reply:
Added in [`a31d5f6` (#733)](https://github.com/chainguard-dev/malcontent/pull/733/commits/a31d5f6aaa4dfbe494a6eccc016e0dc9e6751083).